tfjs-backend-webgpu/src/index.ts

```diff
@@ -37,13 +37,15 @@ registerBackend('webgpu', async () => {
 
   const adapter = await navigator.gpu.requestAdapter(gpuDescriptor);
   let deviceDescriptor: GPUDeviceDescriptor = {};
-  const supportTimeQuery =
-      adapter.extensions.includes('timestamp-query' as GPUExtensionName);
+  const supportTimeQuery = adapter.features.has('timestamp-query');
 
   if (supportTimeQuery) {
-    deviceDescriptor = {extensions: ['timestamp-query' as const ]};
+    deviceDescriptor = {
+      nonGuaranteedFeatures: ['timestamp-query' as GPUFeatureName]
```

Comment:
nit: You can still use `as const` here, like before. It is safe against typos, unlike `as GPUFeatureName`.
```suggestion
      nonGuaranteedFeatures: ['timestamp-query' as const]
```

Reply:
> nit: You can still use `as const` here, like before. It is safe against typos, unlike `as GPUFeatureName`.

Got it. Thanks.